### DISCUSSION TITLE
Reanimated matchers for Jest in TypeScript

### DISCUSSION THREAD

**Discussion by hamidhadi** [software-mansion/react-native-reanimated#3127] [Q&A]
I'm trying to test a component style with Jest & testing-library.
All my tests are in TS and the problem is Reanimated matchers are not supported by Jest types so this line raises an error:
```ts
expect(MyComponent).toHaveAnimatedStyle(expectedStyle); // <<  Property 'toHaveAnimatedStyle' does not exist on type 'JestMatchers<ReactTestInstance>'
```
I have a feeling that it must be something obvious and I'm missing something 😅 but I couldn't find anything in the doc about it.
It would be nice if someone can help me. 🙏

**Comment by Stewartarmbrecht** [comment]
Did you ever figure this out?  I am having the same problem.

**Comment by MatiPl01** [comment]
Have you added this line to your `jest-setup.js` file?

```js
require('react-native-reanimated').setUpTests();
```

(see [docs](https://docs.swmansion.com/react-native-reanimated/docs/guides/testing/#setup) for more details).

**Comment by MatiPl01** [comment]
It looks like this discussion is stale as there are no more new questions.

I am closing the discussion. If you still have questions, feel free to re-open the discussion or start a new one.
